test Foodsynths

Poojawa opened the pull request:
god I hate UIs so much lmfao

Poojawa committed:
Initial re-port

Passes DM Compile test at least! Tgui menu hella broke still.

Poojawa committed:
how did it become 12 hours later

uhhhhhhhhg

UI backend almost working. Needs icon mischief to be managed. Don't want another GetFlatIcon incident.

Poojawa committed:
More tweaks before a DEMOCRACY break

Poojawa committed:
sorting woes

Poojawa committed:
lmao how does this still compile

Poojawa committed:
let's see if the checks will pass

Poojawa committed:
Shift recipes to correct place

Also Prettier Tgui says my UI is ok. so everything compiles :D

Poojawa committed:
strip out all the crew stuff AGAIN

also actually get some useful info if we're spawning the item. Because, yeah?

Poojawa committed:
Maybe actually make the args match??????

Turns out, making a cookie TRUE/FALSE isn't good for your database keys

Poojawa committed:
return, my son, to Autolathe code again

Stop being complex, you don't need to be.

Poojawa committed:
typical, tbh.

Poojawa committed:
yup, this too. smh

Poojawa committed:
Check if Not, not if do.

sigh.

Poojawa committed:
if only there was some way to debug easier

Poojawa commented:
TypeError: Unable to get property 'prototype' of undefined or null reference
   at Anonymous function (http://127.0.0.1:63357/tmp50328/tgui.bundle.js:1:157639)
   at t.createComponentVNode (http://127.0.0.1:63357/tmp50328/tgui.bundle.js:1:157582)
   at e (http://127.0.0.1:63357/tmp50328/tgui.bundle.js:1:616515)
   at n (http://127.0.0.1:63357/tmp50328/tgui.bundle.js:1:1295393)
   at r (http://127.0.0.1:63357/tmp50328/tgui.bundle.js:1:1294199)
   at t.getRoutedComponent (http://127.0.0.1:63357/tmp50328/tgui.bundle.js:1:1258746)
   at Anonymous function (http://127.0.0.1:63357/tmp50328/tgui.bundle.js:1:1297931)
   at Anonymous function (http://127.0.0.1:63357/tmp50328/tgui.bundle.js:1:1256981)
   at a (http://127.0.0.1:63357/tmp50328/tgui.bundle.js:1:197945)
   at Anonymous function (http://127.0.0.1:63357/tmp50328/tgui.bundle.js:1:218819)
User Agent: Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.2; WOW64; Trident/7.0; .NET4.0C; .NET4.0E; .NET CLR 2.0.50727; .NET CLR 3.0.30729; .NET CLR 3.5.30729; Tablet PC 2.0; Creative AutoUpdate v1.41.09)
State: {"ckey":"poojawa","interface":"FoodSynthesizer","window":{"key":"[0x2014911]-main","size":null,"fancy":1,"locked":0}}

Poojawa committed:
Turns out, Flexing doesn't work for me or the code

Poojawa committed:
Ballmer peak??????

Poojawa committed:
no no no, we shove this INTO

Stop having it  be hanging out. finnicky as fuck dev tool

Poojawa committed:
MENU WORKING

BUT IT DOESN'T WANT TO POSITION ITSELF AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

Poojawa committed:
Still working on layout positions...

how do I get details shifted over properly aaaaa

Poojawa committed:
compile tgui like I oughta

Poojawa commented:
for fuck's sake why does github decide I need ALL of the other commits added...

Poojawa committed:
maybe this'll populate my menus

also cuts down on the menu bloat from the .js file. or should anyway

Poojawa committed:
I can either populate my tabs or menus

For whatever reason, I just can't do both without a FUCKLOAD of copypasta.

Ughsdfuighfduighfudhg

Poojawa committed:
still no closer. godamnit.

Poojawa committed:
really just need byond to work with me here

Arokha commented on code/modules/food/recipes_foodsynthesizer.dm:
```diff
@@ -0,0 +1,1114 @@
+	..()
+	var/obj/item/weapon/reagent_containers/food/snacks/snacc
+	if(path)
+		snacc = new path()
```
You don't actually need to create an instance of the item to use initial() on it a few lines down. You could avoid some performance overhead by just calling initial on the path itself, which byond allows for most vars (including the name and desc).

Poojawa replied:
This is how it works for autolathe code, trying to mimic R&D's stuff auto fails because designs aren't category items

Poojawa committed:
w e h

Poojawa committed:
setactive_menu called with appasnacc x1223423

Poojawa committed:
WHY ISN'T IT POSSIBLE?

Javascrip: "It's just not."

WHY NOT, YOU STUPID BASTARD

Poojawa committed:
Let's SetState to be fancy with vars

Poojawa committed:
Code Sune SOS answered tabs work

Let's figure out why my recipes aren't anymore

Poojawa committed:
Bro all it needs to do is fucking self update

why the fuck is this so hard to do?

Poojawa committed:
it WOULD be a single word mispell.

Why WOULDN'T it be something so simple I'd been squinting at.

Poojawa committed:
it's woooorking :D

Poojawa committed:
Beginnings of crew cookies again

this time with more plausibility! Also preview icons as well

Poojawa committed:
Apparently mobs don't have references. weird.

Poojawa committed:
Fixes some cosmetic issues

still fiddling with adding images to the UI. bweh. Least the interactions work! E/W sprites also got tweaked sprites to help distinguish status when facing those directions, just in case. You'll be able to see if it's working or broken!

Poojawa committed:
How do spritesheets work, ffs

Poojawa committed:
Spritesheets working

We have actual pictures in the menu now!!!

Now to struggle and scream about crew related things hhhhhh

Poojawa committed:
why doesn't flex wrapping work

Poojawa committed:
fiddling with layout stuff

Poojawa committed:
Okay well, crew stuff sort of working?

Except for whatever reason it isn't getting the entire icon information... Or it is, but icon2base64 isn't serializing the entire icon

Or...something idfk

Poojawa committed:
m e h

Poojawa committed:
FULLY WORKING

Poojawa committed:
oops, we might actually want menus tbh

Poojawa committed:
Make them actually small
